tests/integration/utils.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+
+declare -g CONTAINER_DIR # has the container files to build the container images
+declare -g SAMPLES_DIR   # has sample files used accross the integration tests
+declare -g KWROOT_DIR    # local kw dir to be copied to and installed in the containers
+declare -g DISTROS       # distributions we will run the integration tests
+
+# ensure path to directories is absolute
+script_dir=$(realpath "$(dirname "${0}")")
+CONTAINER_DIR="${script_dir}/podman"
+SAMPLES_DIR="${script_dir}/samples"
+KWROOT_DIR=$(realpath "${script_dir}/../..")
+
+# supported distros
+DISTROS=('archlinux' 'debian' 'fedora')
+
+# Builds a container image for the given distro
+function build_distro_image()
+{
+  local distro="${1}"
+  local file="${CONTAINER_DIR}/Containerfile_${distro}"
+
+  podman image build --file "$file" --tag "kw-${distro}" 2> /dev/null
+
+  # Check if the command failed
+  if [[ $? -ne 0 ]]; then
+    fail "($LINENO): Error building the image for distribution ${distro}"
+  fi
+}
+
+# Builds container images and create containers used accross the tests.
+function setup_container_environment()
+{
+  local container_img
+  local container_name
+  local distro
+  local distros_ok=() # array of distros whose setup succeed
+  local sha_container
+  local sha_kw
+  local working_directory
+  local output # output of podman used to check if containers exist or not
+
+  for distro in "${DISTROS[@]}"; do
+    # Only build the image if it does not exists. That's because trying to build
+    # the podman image takes a second or two even if it exists and is cached.
+    podman image exists "kw-${distro}"
+    if [[ $? -ne 0 ]]; then
+      # Build the image or exit. The integration tests cannot continue otherwise
+      build_distro_image "$distro" > /dev/null
```

Comment:
Great idea! Will do.

Reply:
@rodrigosiqueira can you please detail how this verbose feature should work? 
For example, should podman's output be shown to the user while it builds the container images? Which information should be displayed (maybe all?) and which should be supressed?

Just a side note: Initially, I intetionally redirect all of podman's output to /dev/null because I do not like to see it mixed with the default standardized output of tests (that's my personal preference).

Reply:
For the verbose mode, let's show everything.